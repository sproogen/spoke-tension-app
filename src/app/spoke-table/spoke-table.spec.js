import React from 'react'
import { shallow } from 'enzyme'
import SpokeTable from './spoke-table'

jest.mock('../conversions', () => ({
  getTension: jest.fn(() => 100),
}))

const props = {
  spokeCount: 16,
  toolId: 'tool',
  spokeId: 'spoke-1.8',
}

describe('SpokeRow', () => {
  describe('render', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<SpokeTable {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
