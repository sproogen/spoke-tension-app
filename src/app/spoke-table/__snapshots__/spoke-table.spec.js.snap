// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpokeRow render should match snapshot 1`] = `
<WithStyles(Paper)>
  <WithStyles(Table)>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)>
          Spoke #
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Reading
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Tension (kgf)
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Variance (%)
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 0,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 1,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 2,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 3,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 4,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 5,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 6,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 7,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 8,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 9,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 10,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 11,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 12,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 13,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 14,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
      <SpokeRow
        averageTension={100}
        key="spoke-undefined"
        spoke={
          Object {
            "number": 15,
            "reading": "",
            "spokeId": "spoke-1.8",
            "toolId": "tool",
          }
        }
        updateReading={[Function]}
      />
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</WithStyles(Paper)>
`;
